### PR TITLE
chore: add architecture param for EMR application

### PR DESCRIPTION
**Pull request by amliuyong**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary
- add `EmrApplicationArchitecture` parameter for data-pipeline stack
- 'X86_64' as the default value for the parameter in China regions
- 'ARM64' as the default value for the parameter in regions except China

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [x] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [x] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: src/data-pipeline/parameter.ts [hunk 1]
```diff
@@ -135,6 +135,17 @@ export function createStackParameters(scope: Construct) {
     type: 'Number',
   });
 
+  const emrApplicationArchitectureParam = new CfnParameter(scope, 'EmrApplicationArchitecture', {
+    description: 'Emr-Serverless application architecture',
+    default: 'Auto',
+    allowedValues: [
+      'Auto',
```
Comment (zxkane) [inline src/data-pipeline/parameter.ts:142]: Let's move those as constants to share with stack.

Also, could you reuse those architect definitions of the AWS SDK?

File: src/data-pipeline/lambda/emr-serverless-app/index.ts [hunk 2]
```diff
@@ -75,11 +77,23 @@ async function _handler(event: CloudFormationCustomResourceEvent, context: Conte
 }
 
 async function createEMRServerlessApp(props: ResourcePropertiesType): Promise<string> {
+  let architecture: Architecture = Architecture.X86_64;
+  if (props.architecture === 'Auto') {
+    architecture = Architecture.ARM64;
+    if (region.startsWith('cn-')) {
+      architecture = Architecture.X86_64;
+    }
+  } else if (props.architecture === 'ARM64') {
```
Comment (zxkane) [inline src/data-pipeline/lambda/emr-serverless-app/index.ts:86]: use the architecture in SDK if possible

**Review by zxkane — CHANGES_REQUESTED** [review]

File: src/data-pipeline/parameter.ts (hunk 1 above)
Comment (amliuyong) [reply src/data-pipeline/parameter.ts]: changed.

File: src/data-pipeline/lambda/emr-serverless-app/index.ts (hunk 2 above)
Comment (amliuyong) [reply src/data-pipeline/lambda/emr-serverless-app/index.ts]: changed.

**Review by zxkane — APPROVED** [review]

**Pull request merged**